app/learning.js

```diff
@@ -1,33 +1,56 @@
 /* file: learning.js
- * authors: H. Bouakaz, S. de Vliet, S. de Jong & E. Werkema
- * date: 5/3/2016
- * version 1.1
+ * authors: H. Bouakaz, S. van Vliet, S. de Jong & E. Werkema
+ * date: 25/3/2016
+ * version 1.2
  *
  * Description:
  * Main script for initiating the learning app.
  */
 
-define(['jquery', 'bootstrap', 'app/config', 'app/database', 'app/messages', 'app/question', 'app/timer'], function ($, bootstrap, config, db, messages, question, timer) {
-  timer.startTimer(config.constant("TIME_LIMIT"));
+define(['jquery', 'bootstrap', 'app/config', 'app/database', 'app/messages', 'app/question', 'app/timer','app/database'], function ($, bootstrap, config, db, messages, question, timer,db) {
+  timer.startTimer(".timer", config.constant("TIME_LIMIT"));
+  var waitingForEnter = false;
 
   // Disable autocomplete that provides suggestions when typing words
   $('input').attr('autocomplete', 'off');
-
+//when the page is loaded we get the datasetId from the page url and load the dataset from the databese
   $(document).ready(function() {
+    var url =   window.location.href
```

Comment:
This seems impractical, what other methods have you considered here and why did you drop them?